### DISCUSSION TITLE
Union hangs with junction & reused flatSurface's

### DISCUSSION THREAD

**Discussion by jlvila** [jimy-byerley/pymadcad#34] [Q&A] [answered]
Dear,

Here is a piece of code that produces a `TriangulationError `(line of code : `_p2 = union(_cone, _p1)`). 
That is specific is that the the `_cone `is built using 2 two `interfaces `that I want to reuse to have a closed volume.

```python
from madcad import *
x = 1
y = 1
w = 1
h = 1

_b2 = Box(vec3(0, 0, 0), vec3(
    x+w, y+h, 10))
_br2 = brick(_b2)

color = vec3(1)

_radius = w/2.0
_length = h
interfaces = [
    flatsurface(wire(Circle((vec3(0), vec3(0, 1, 0)), _radius))),
    flatsurface(wire(Circle((vec3(0, _length, 0), vec3(0, -1, 0)), 0.1)))
]

_cone = junction(
    interfaces[0],
    interfaces[1],
    tangents='straight'
)
_cone.mergeclose()
_p1 = union(interfaces[0], interfaces[1])
_p1.mergeclose()
_p2 = union(_cone, _p1)
show([_p2])
```

**Comment by jimy-byerley** [comment, marked as answer]
Hi @jlvila

It's true that the boolean operations gets touchy when one of the two surfaces of the operation does intersect the second only on one of its frontiers (like in your case where the interfaces only intersects the cone by their outer ring). That's a common issue with #8

Luckyly in your case (If I correctly understood) you don't need boolean operations with intersection, stiching, etc. You can simply concatenate your cone surface with its ends surfaces using the `+` operator:

```python
_cone = junction(
    interfaces[0],
    interfaces[1],
    tangents='straight'
)
_cone.mergeclose()

_p2 = _cone + interfaces[0].flip() + interfaces[1].flip()
_p2.mergeclose()
```

**Comment by jlvila** [comment]
@jimy-byerley Many thanks again for your work,

**Reply by jimy-byerley** [reply]
With great pleasure :) I hope it is of some help for you !
